test/unit/adapters/http.js

```diff
@@ -17,484 +26,442 @@ module.exports = {
     if (process.env.http_proxy) {
       delete process.env.http_proxy
     }
+  })
 
-    callback()
-  },
+  it('does handle timeouts', async () => {
+    nock(testUri)
+      .get('/')
+      .delayConnection(1000)
+      .reply(200, '<html></html>')
 
-  testTimeout: function (test) {
-    server = http.createServer(function (req, res) {
-      setTimeout(function () {
-        res.end()
-      }, 1000)
-    }).listen(4444, function () {
-      let success = false, failure = false
-      let error
-
-      axios.get('http://localhost:4444/', {
+    try {
+      await axios.get(testUri, {
```

Comment:
Using `async` `await` will require us to use different configs than the library which may introduce some confusion.
It would be nice to use `Promises` for now. To align the feature use across the library